skylog/core/builder_test.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/google/trillian/merkle/compact"
+	"github.com/google/trillian/merkle/rfc6962"
+	"github.com/google/trillian/merkle/testonly"
+	"github.com/google/trillian/skylog/storage"
+)
+
+// TODO(pavelkalinnikov): Use GoMock?
+type mockTreeWriter struct {
+	calls [][]storage.Node
+}
+
+func (t *mockTreeWriter) Write(ctx context.Context, nodes []storage.Node) error {
+	t.calls = append(t.calls, nodes)
+	return nil
+}
+
+func (t *mockTreeWriter) verify(nodes []storage.Node) error {
+	if len(t.calls) == 0 && len(nodes) == 0 {
+		return nil
+	}
+	if got, want := len(t.calls), 1; got != want {
+		return fmt.Errorf("wrong number of calls %d, want %d", got, want)
+	}
+	if got, want := t.calls[0], nodes; !reflect.DeepEqual(got, want) {
+		return fmt.Errorf("got nodes: %+v\nwant: %+v", got, want)
+	}
+	return nil
+}
+
+func TestBuildWorker(t *testing.T) {
+	hashes := testonly.NodeHashes()
+	n := len(hashes[0])
+	if got, want := n, 8; got != want {
+		t.Fatalf("got %d leaves, want %d", got, want)
```

Comment:
nit: maybe change the logged message here - "prerequisite: testinly.NodeHashes() returned %d leaves, but this test wants 8" - just to make it clear that it's not the _test per se_ which has failed.

Reply:
Done.